base/strings/io.jl

```diff
@@ -69,15 +69,21 @@ println(io::IO, xs...) = print(io, xs..., '\n')
 ## conversion of general objects to strings ##
 
 """
-    sprint(f::Function, args...)
+    sprint(f::Function, args...; context=nothing, sizehint::Integer=0)
 
 Call the given function with an I/O stream and the supplied extra arguments.
 Everything written to this I/O stream is returned as a string.
+`context` can be either an [`IOContext`](@ref) whose properties will be used,
+or a `Pair` specifying a property and its value. `sizehint` suggests the capacity
```

Comment:
Specifying a Pair does happen to work, but only the usage case of passing an IO was what I had really originally intended (and found to be common).

I think that we maybe instead should suggest the usage of anonymous functions as a general way of passing a configuration set of arbitrarily many new properties:
```julia
showcompact = (io, v) -> show(IOContext(io, :compact => true), v)
sprint(showcompact, 66.66666)
```

Reply:
OK. I assumed it was intended since  the signature doesn't use `::IOContext`.

I actually started with the anonymous function approach, but it's quite verbose and much less convenient than passing a pair for common cases (have a look at the diff to get and idea). So I'd say we'd better continue to support passing a pair.

Reply:
I don't intend to prevent passing a Pair, but it simply doesn't generalize as well. Although I suppose it also quickly becomes easier just to write it out:
```julia
buf = IOBuffer(sizehint = sizehint)
io = IOContext(buf, :compact => true)
show(io, 66.66666)
return String(take!(buf))
```

Reply:
My thought was that passing a single pair is the most common case, and for other situations it's not too hard to create a full `IOContext`.

Reply:
I agree it's very nice to be able to pass a single pair.